app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,28 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    # check to see if already in queue
+    already_in_queue = current_user.queue_items.any? {|item| item.video_id == params[:video_id].to_i} 
```

Comment:
Another way to do this without pulling all of the items into memory is using `pluck`.

```ruby
already_in_queue = current_user.queue_items.pluck(:video_id).include? params[:video_id].to_i
```